osu.Game/Overlays/Mods/DeselectAllModsButton.cs

```diff
@@ -18,13 +18,16 @@ namespace osu.Game.Overlays.Mods
     public partial class DeselectAllModsButton : ShearedButton, IKeyBindingHandler<GlobalAction>
     {
         private readonly Bindable<IReadOnlyList<Mod>> selectedMods = new Bindable<IReadOnlyList<Mod>>();
+        private readonly ShearedSearchTextBox searchTextBox;
 
         public DeselectAllModsButton(ModSelectOverlay modSelectOverlay)
             : base(ModSelectOverlay.BUTTON_WIDTH)
         {
             Text = CommonStrings.DeselectAll;
             Action = modSelectOverlay.DeselectAll;
 
+            searchTextBox = modSelectOverlay.SearchTextBox;
```

Comment:
Not only is this pretty bad coupling, this PR also doesn't solve the same problem with the *other* button that has this problem, namely `SelectAllModsButton`.

- Go to free mods overlay
- Focus search box, type anything (but something that keeps some mods visible)
- <kbd>Ctrl</kbd>+<kbd>A</kbd>
- Observe as text in the search box is selected at the same time as all visible mods are also selected.

The handling of these hotkeys should be moved out to the mod overlay itself. It should be handling both `GlobalAction`s and calling `TriggerClick()` on both buttons if search textbox isn't focused. This gets rid of the weird coupling here, and of the need to expose `HasFocus` on the sheared textbox.

Reply:
`SelectAllModsButton` uses `PlatformAction.SelectAll` (which means select all text if I assume correctly), so should I change the action or block SelectAll for `ShearedSearchTextBox`?

Reply:
For the sake of everyone's collective sanity, please just block for now. Changing it is an entire separate discussion (why would it need changing? if it needs changing, what should it be changed to? etc etc)